core/modules.rs

```diff
@@ -321,7 +325,7 @@ pub struct Deps {
 }
 
 impl Deps {
-  pub fn new(modules: &Modules, module_name: &str) -> Deps {
+  fn new(modules: &Modules, module_name: &str) -> Option<Deps> {
```

Comment:
If this function returns an Option<Deps> then I think the field Deps::deps shouldn't be an Option any more.